provider/aws.go

```diff
@@ -30,7 +30,30 @@ import (
 )
 
 const (
-	hostedZonePrefix = "/hostedzone/"
+	hostedZonePrefix     = "/hostedzone/"
+	elbHostnameSuffix    = ".elb.amazonaws.com"
+	evaluateTargetHealth = false
+	recordTTL            = 300
+)
+
+var (
+	// see: https://docs.aws.amazon.com/general/latest/gr/rande.html
+	canonicalHostedZones = map[string]string{
+		"us-east-1":      "Z35SXDOTRQ7X7K",
```

Comment:
why not just store whole suffix: 
`"us-east-1.elb.amazonaws.com":"Z35SXDOTRQ7X7K"...` 

and then if CanonicalZoneID function does suffix matching and returns corresponding zoneId. if zoneId is "" then create a CNAME ?